src/cluster.c

```diff
@@ -5591,6 +5565,41 @@ NULL
                 (retval == C_OK) ? "BUMPED" : "STILL",
                 (unsigned long long) myself->configEpoch);
         addReplySds(c,reply);
+    } else if (!strcasecmp(c->argv[1]->ptr,"slot-stats")) {
+        unsigned char *slots = zmalloc(CLUSTER_SLOTS);
+        memset(slots, 0, CLUSTER_SLOTS);
+        if (c->argc == 2) {
+            /* CLUSTER SLOT-STATS */
+            markAssignedSlots(slots);
+            addReplySlotStats(c, slots);
+        } else if (c->argc >= 3 && !strcasecmp(c->argv[2]->ptr,"slots")) {
+            /* CLUSTER SLOT-STATS SLOTS slot [slot ...]*/
+            if (checkSlotStatsSlotsArgumentOrReply(c, slots) == C_ERR) {
+                zfree(slots);
+                return;
+            }
+            addReplySlotStats(c, slots);
+        } else if (c->argc >= 3 && !strcasecmp(c->argv[2]->ptr,"slotsrange")) {
+            /* CLUSTER SLOT-STATS SLOTSRANGE start-slot end-slot [start-slot end-slot ...] */
+            if (checkSlotStatsSlotsRangeArgumentOrReply(c, slots) == C_ERR) {
+                zfree(slots);
+                return;
+            }
+            addReplySlotStats(c, slots);
+        } else if (c->argc >= 4 && !strcasecmp(c->argv[2]->ptr,"orderby")) {
```

Comment:
I think it would be easier to follow if we had just one parser for arguments. Most other places we incrementally move through the arguments and update values based on them. I would look at something like https://github.com/redis/redis/blob/unstable/src/cluster.c#L6062 as a good example for parsing all arguments. I also think the specific functions for parsing ORDERBY and SLOTSRANGE is weird, if they are mutually exclusive we should have an error indicating that they are incompatible.

Reply:
Probably antirez liked the letter j. :-) Use i or j as you want. it's not regulated by our guidelines.

Reply:
@zuiderkwast Ok, `int i` will do then :-)

@madolson On second thought, I disagree on incrementally moving through the arguments, for **mutually exclusive** arguments, such as `SLOTSRANGE` and `ORDERBY`.

For mutually inclusive args, incremental iteration is ideal, as optional arguments can come and go at any ordering permutations. You simply can not expect it to show at index 2 or 5 ... and so on. 

However, for mutually exclusive items, they MUST fall under a specified index. 2nd index for `CLUSTER SLOT-STATS <mutually_exclusive_arg>` command. Once the mutually exclusive argument is parsed, the code should then proceed further with incremental iteration for optional arguments under the mutually exclusive arguments.

The standard I suggest is the following:
```

```

I've identified multiple examples of this case. The shared `RESTORE` command is not ideal, since it contains only .

XGROUP command is a better example, where the following subcommands are all mutually exclusive, and thus are dealt